src/linear_samplers/block_col_sparse_sign.jl

```diff
@@ -0,0 +1,107 @@
+# This file is part of RLinearAlgebra.jl
+
+
+"""
+    LinSysBlkColSparseSign <: LinSysBlkColSampler
+
+A mutable structure with fields to handle sparse sign column sketching where a sparse sign matrix
+is multiplied by the matrix `A'` from the left. More details of the methods are mentioned in the 
+section 9.2 of 
+
+Martinsson P-G, Tropp JA. "Randomized numerical linear algebra: Foundations and algorithms." 
+Acta Numerica. 2020;29:403-572. doi:10.1017/S0962492920000021.
+
+# Fields
+- `block_size::Int64`, represents the number of rows in the sketch matrix.
+- `sketch_matrix::Union{AbstractMatrix, Nothing}`, buffer for storing the sparse sign sketching matrix.
+- `numsigns::Int64`, buffer for storing how many signs we need to have for each column of the 
+    sketch matrix, which can be chosen from `{2, 3, ..., block_size}.
+- `scaling::Float64`, the standard deviation of the sketch, set to `sqrt(n / numsigns)`.
+
+# Constructors
+- `LinSysBlkColSparseSign()` defaults to setting `block_size` to 8 and `numsigns` to `min(block_size, 8)`.
+"""
+mutable struct LinSysBlkColSparseSign <: LinSysBlkColSampler
+    block_size::Int64
+    numsigns::Union{Int64, Nothing}
+    sketch_matrix::Union{Matrix{Int64}, Nothing}
+    scaling::Float64
+    LinSysBlkColSparseSign(block_size, 
+                           numsigns, 
+                           sketch_matrix, 
+                           scaling) = begin
+        check_properties = (block_size > 0)
+        @assert check_properties "`block_size` must be positive."
```

Comment:
No need to assign new variable `check_properties`. You can replace this with the expression from the previous line.